llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -2826,8 +2826,16 @@ HandleMergeInputChains(SmallVectorImpl<SDNode*> &ChainNodesMatched,
   // node that is both the predecessor and successor of the
   // to-be-merged nodes. Fail.
   Visited.clear();
-  for (SDValue V : InputChains)
+  for (SDValue V : InputChains) {
+    // If we need to create a TokenFactor, and any of the input chains will
```

Comment:
(nit) "input chains will also be glued" doesn't read right. It is nodes that are glued, not chains